### PR TITLE
Add tags public form

### PR DESCRIPTION
**Pull request by Bjoern-Jentschke**
added tags to the recipe form. Just one tag can be chosen. The tag can also be edited. In the edited form all the default values are set based on the values in the database. There is a toggle checkbox to declare in the recipe form if the recipe is private or public.

### CODE REVIEW

File: components/RecipeForm.js [hunk 1]
```diff
@@ -33,27 +32,36 @@ export default function RecipeForm({ onSubmit, onDelete, data }) {
...
+    const isAlreadySelected = selectedTags.includes(label);
+    let newTags = label;
+    if (isAlreadySelected) {
+      newTags = selectedTags.filter((item) => item !== label);
+      setSelectedTags(newTags);
+    } else {
+      newTags = [label];
+      setSelectedTags(newTags);
+    }
+  }
```
Comment (MerleMak) [inline components/RecipeForm.js]:
```suggestion
  function handleTagChange(label) {
    setSelectedTags(selectedTags.includes(label) 
    ? selectedTags.filter(item => item !== label) 
    : [label]);
  }
```

File: components/RecipeForm.js [hunk 2]
```diff
@@ -33,27 +32,36 @@ export default function RecipeForm({ onSubmit, onDelete, data }) {
...
-    }
+    const uploadResponse = await fetch(
+      "https://api.cloudinary.com/v1_1/mealwheel/image/upload",
+      {
+        method: "POST",
+        body: data,
+      }
+    );
+    const file = await uploadResponse.json();
+    setImageUrl(file.secure_url);
```
Comment (MerleMak) [inline components/RecipeForm.js]: this seems like it shouldn't be part of this US?

**Review by MerleMak — COMMENTED** [review]
Looks great! 
I left you one simplification of a function. 

Nice job! Keep it up :)

File: components/RecipeForm.js (hunk 2 above)
Comment (Bjoern-Jentschke) [reply components/RecipeForm.js]: Thank you very much for the comment. That was part of the [https://github.com/users/gcode-de/projects/1/views/1?pane=issue&itemId=55384387](userStory - addNewRecipe - imageAdding). The code was implemented and written one week ago. I cannot remember changing something in this function. So indeed not a part of this userStory.:)

**Comment by Bjoern-Jentschke** [comment]
Thank you!!!:))

**Pull request merged**